sites/kit.svelte.dev/src/routes/home/Video.svelte

```diff
@@ -0,0 +1,128 @@
+<script>
+	import testimonials from './testimonials.mp4';
+	import volume_off from './volume-off.svg';
+	import volume_high from './volume-high.svg';
+	import { onMount } from 'svelte';
+
+	/** @type {HTMLVideoElement} */
+	let video;
+	let muted = true;
+	let unused = true;
+
+	let d = 0;
+	let t = 0;
+
+	onMount(() => {
+		// I think this binding is broken in SSR because the video already
+		// has a duration by the time hydration occurs. TODO investigate
+		d = video.duration;
+
+		const observer = new IntersectionObserver(
+			(entries) => {
+				for (const entry of entries) {
+					if (entry.isIntersecting) {
+						video.play();
+						observer.disconnect();
+					}
+				}
+			},
+			{
+				threshold: 1
+			}
+		);
+
+		observer.observe(video);
+
+		return () => {
+			observer.disconnect();
+		};
+	});
+</script>
+
+<div class="video-player">
+	<video
+		src={testimonials}
+		loop
+		bind:this={video}
+		bind:muted
+		bind:currentTime={t}
+		bind:duration={d}
+		on:click={() => {
+			if (video.paused) {
+				video.play();
+			} else {
+				video.pause();
+			}
+		}}
+	>
+		<!-- TODO-->
+		<track kind="captions" />
```

Comment:
Flagging this to make sure we don't forget about captions.

Reply:
done (thanks for the Whisper tip @kevmodrome). Not sure if it should be visible by default, given that the audio isn't playing at first. Probably not unless we add UI for toggling it off

Reply:
Won't we need toggle UI anyway? Since if they're not visible by default, then we'd need a way to enable them.

IMO I would have them visible by default, since I often have an aversion to playing sound on a video (e.g. when scrolling Twitter). It would be nice for people to see what the video is covering without needing to enable sound or figure out that a caption control is available.